fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -189,25 +203,24 @@ public void writeData(long id, int block, @Nonnull byte[] value) {
      * @param referenceFuture the reference where the data supposedly lives
      * @param block the block where the data is stored
      * @return Completable future of the data returned
-     * @throws IOException if blockCache fails to get the data from the block
+     * @throws RecordCoreException if blockCache fails to get the data from the block
      * @throws NullPointerException if a reference with that id hasn't been written yet.
      */
     @Nonnull
-    public CompletableFuture<byte[]> readBlock(@Nonnull String resourceDescription, @Nonnull CompletableFuture<FDBLuceneFileReference> referenceFuture, int block) throws IOException {
+    public CompletableFuture<byte[]> readBlock(@Nonnull String resourceDescription, @Nonnull CompletableFuture<FDBLuceneFileReference> referenceFuture, int block) throws RecordCoreException {
         try {
             LOGGER.trace("readBlock resourceDescription={}, block={}", resourceDescription, block);
             final FDBLuceneFileReference reference = referenceFuture.join(); // Tried to fully pipeline this but the reality is that this is mostly cached after listAll, delete, etc.
             Long id = reference.getId();
             return blockCache.get(Pair.of(id, block),
                     () -> {
-                        Long value = reads.getOrDefault(resourceDescription + ":" + id, 0L);
-                        value += 1;
-                        reads.put(resourceDescription + ":" + id, value);
-                    return context.ensureActive().get(dataSubspace.pack(Tuple.from(id, block)));
-                }
+                        return context.ensureActive().get(dataSubspace.pack(Tuple.from(id, block)));
+                    }
             );
         } catch (ExecutionException e) {
-            throw new IOException(e);
+            throw new RecordCoreException("Execution exception thrown while getting block cache", e.getCause());
```

Comment:
Apparently there is a convention for the record layer
```suggestion
            throw new RecordCoreException(CompletionExceptionLogHelper.asCause(e));
```